src/main/java/edu/cornell/library/integration/hathitrust/BuildHathiFilesDiffReport.java

```diff
@@ -0,0 +1,129 @@
+		List<String> dbPrefixes = identifyPrefixes(env.get("prefixes"));
+		System.out.println("Prefixes: '"+String.join("', '", dbPrefixes)+"'");
+		if (dbPrefixes.size() != 2) {
+			System.out.println("Exactly 2 prefixes must be specified for comparison.");
+			System.exit(1);
+		}
+		confirmTablesExist(config, dbPrefixes);
+		Set<String> ids = getAllVolumeIdentifiers(config, dbPrefixes);
+		compareRecords(config, dbPrefixes, ids);
+	}
+
+	private static void compareRecords(Config config, List<String> prefixes, Set<String> ids) throws SQLException {
+		String table1 = prefixes.get(0) + "raw_hathi";
+		String table2 = prefixes.get(1) + "raw_hathi";
+		try (Connection hathidb = config.getDatabaseConnection("Hathi");
+				PreparedStatement stmt1 = hathidb.prepareStatement(
+						String.format("SELECT * FROM %s WHERE Volume_Identifier = ?", table1));
+				PreparedStatement stmt2 = hathidb.prepareStatement(
+						String.format("SELECT * FROM %s WHERE Volume_Identifier = ?", table2));
+				) {
+			for (String id : ids) {
+				stmt1.setString(1, id);
+				stmt2.setString(1, id);
+				try (ResultSet rs1 = stmt1.executeQuery(); ResultSet rs2 = stmt2.executeQuery()) {
+					if (! rs1.next()) { System.out.printf("%s not in table %s\n", id, table1); continue; }
+					if (! rs2.next()) { System.out.printf("%s not in table %s\n", id, table2); continue; }
+					List<String> differences = new ArrayList<>();
+					String[] stringFields = {"Access", "Rights","UofM_Record_Number","Enum_Chrono","Source",
+							"Source_Inst_Record_Number","OCLC_Numbers","ISBNs","ISSNs","LCCNs","Title","Imprint",
+							"Rights_determine_reason_code","Date_Last_Update","Pub_Date","Pub_Place","Language",
+							"Bib_Format","Digitization_Agent_code","Content_provider_code","Responsible_Entity_code",
+							"Collection_code","Access_profile","Author"};
+					for (String field : stringFields) {
+						String val1 = rs1.getString(field);
+						String val2 = rs2.getString(field);
+//						System.out.printf("%s %s : %s => %s\n", id, field, val1, val2);
+						if (! Objects.equals(val1, val2))
+							differences.add(String.format("%s: %s => %s", field, val1, val2));
+					}
+					String[] intFields = {"Gov_Doc"};
+					for (String field : intFields) {
+						Integer val1 = rs1.getInt(field);
+						Integer val2 = rs2.getInt(field);
+						if (! Objects.equals(val1, val2))
+							differences.add(String.format("%s: %d => %d", field, val1, val2));
+					}
+					if ( ! differences.isEmpty() )
+						System.out.printf("%s: [%s]\n", id, String.join("; ", differences));
+				}
+			}
+		}
+	}
+
+
+	private static Set<String> getAllVolumeIdentifiers(Config config, List<String> prefixes) throws SQLException {
+		Set<String> ids = new TreeSet<>();
+		try (Connection hathidb = config.getDatabaseConnection("Hathi");
+				Statement stmt = hathidb.createStatement()) {
+			String table1 = prefixes.get(0) + "raw_hathi";
+			String table2 = prefixes.get(0) + "raw_hathi";
```

Comment:
Are table1 and table2 supposed to be same?

Reply:
Good catch. I just committed a fix for this.